centos7-x86_64-vault.repo

```diff
@@ -0,0 +1,27 @@
+[base]
```

Comment:
Instead of trying to remove repos and replace them with identically named one, it is better to disable and add the alternatives with a new name.
E.g., `sed -i '/^\[/a\enabled=0' /etc/yum.repos.d/CentOS-Base.repo` and name them as `C7.9.2009-vault-base` or something (but not `C7.9.2009-base` since that would collide with the naming from `CentOS-Vault.repo`).

If you do that, it should retain the config after a `yum update`.

Reply:
Feel free to push to this PR.